test-tools/issuer-front-end/src/Main.tsx

```diff
@@ -240,68 +313,40 @@ export default function Main(props: WalletConnectionProps) {
 
-                    const registryMetadataReturnValue = JSON.parse(value);
-                    setSmartContractState(registryMetadataReturnValue);
+                const registryMetadataReturnValue = JSON.parse(value);
+                setSmartContractState(registryMetadataReturnValue);
 
-                    const schemaURL = registryMetadataReturnValue.credential_schema.schema_ref.url;
+                const schemaURL = registryMetadataReturnValue.credential_schema.schema_ref.url;
 
-                    fetch(schemaURL)
-                        .then((response) => {
-                            if (response.status !== 200) {
-                                throw new Error(`Status: ${response.status}; StatusText: ${response.statusText}`);
-                            }
-                            return response.json();
-                        })
-                        .then((json) => {
-                            const { properties } = json.properties.credentialSubject.properties.attributes;
-
-                            const attributeSchemaValues: AttributeDetails[] = [];
-                            Object.entries(properties).forEach(([key, obj]) => {
-                                attributeSchemaValues.push({
-                                    tag: key,
-                                    type: (obj as { type: string }).type,
-                                    value: undefined,
-                                    required: Array.from(
-                                        json.properties.credentialSubject.properties.attributes.required
-                                    ).includes(key),
-                                });
-                            });
-                            setFetchingCredentialSchemaError('');
-                            setAttributeSchema(attributeSchemaValues);
-                        })
-                        .catch((e) => {
-                            setAttributeSchema([]);
-                            setFetchingCredentialSchemaError(
-                                `Could not fetch credential schema from smart contract: ${(e as Error).message}`
-                            );
-                        });
-                })
-                .catch((e) => {
-                    setAttributeSchema([]);
-                    setSmartContractState('');
-                    setViewErrorSmartContractState((e as Error).message);
-                });
+                extractFromSchema(schemaURL)
+                    .then((r) => {
+                        setFetchingCredentialSchemaError('');
+                        setAttributeSchema(r);
+                    })
+                    .catch((e) => {
+                        setAttributeSchema([]);
+                        setFetchingCredentialSchemaError(
+                            `Could not fetch credential schema from smart contract: ${(e as Error).message}`
+                        );
+                    });
+            });
```

Comment:
In case we already fail to make a successful RPC call to the smart contract, it would be nice to display the error:

```suggestion
            }).catch((e) => {
                    setAttributeSchema([]);
                    setSmartContractState('');
                    setViewErrorSmartContractState((e as Error).message);
                });
```